### PR TITLE
Updated to alpine 3.5 and install nginx here

### PR DESCRIPTION
**Pull request by ocraviotto**
This PR finally updates master to using alpine and at the same time fixes security issues found in the `alpine` branch by removing dependency on upstream nginx:alpine, thus moving nginx installation here (using most of the original nginx alpine Dockerfile as base), and upgrading to alpine 3.5, as that allows to obtain newer packages that actually fix those security issues. 

NOTE: This does not solve all security issues as reported via docker hub image scan (using the CVE db), but it does fix all the ones directly affecting our project downstream. 

NOTE2: In the future we will split nginx from adding dumb-init and secretary (which should be moved to a more basic base image as it is used by some other base images we use)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,144 @@
...
+		--lock-path=/var/run/nginx.lock \
+		--http-client-body-temp-path=/var/cache/nginx/client_temp \
+		--http-proxy-temp-path=/var/cache/nginx/proxy_temp \
+		--http-fastcgi-temp-path=/var/cache/nginx/fastcgi_temp \
+		--http-uwsgi-temp-path=/var/cache/nginx/uwsgi_temp \
+		--http-scgi-temp-path=/var/cache/nginx/scgi_temp \
+		--user=nginx \
+		--group=nginx \
+		--with-http_ssl_module \
+		--with-http_realip_module \
```
Comment (ilbambino) [inline Dockerfile]: Now that we build nginx, shouldn't we removed not needed modules?
First glance:
- http_dav_module
- http_mp4_module
- http_gunzip_module
- http_random_index_module
- http_secure_link_module
- http_stub_status_module
- http_auth_request_module
- http_xslt_module=dynamic
- http_image_filter_module=dynamic
- http_perl_module=dynamic
- http_geoip_module=dynamic
- mail_ssl_module
- mail

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,144 @@
...
+		--with-http_slice_module \
+		--with-mail \
+		--with-mail_ssl_module \
+		--with-file-aio \
+		--with-http_v2_module \
+		--with-ipv6 \
+	" \
+	&& addgroup -S nginx \
+	&& adduser -D -S -h /var/cache/nginx -s /sbin/nologin -G nginx nginx \
+	&& apk add --no-cache --virtual .build-deps \
```
Comment (ilbambino) [inline Dockerfile]: all this build deps, shouldn't they be removed at the end?

Comment (ocraviotto) [inline Dockerfile]: They are, `&& apk del .build-deps`

File: Dockerfile (hunk 1 above)
Comment (ocraviotto) [inline Dockerfile]: Yes, that's true, we could remove these ones and any other we're not using, I'll have a look...

File: Dockerfile (hunk 2 above)
Comment (ilbambino) [inline Dockerfile]: Opps, I tried to find it, I am blind. Great

Comment (shirkevich) [inline Dockerfile]: If only change from upstream nginx-alpine is upgrade apline version to 3.5 https://github.com/nginxinc/docker-nginx/blob/25a3fc7343c6916fce1fba32caa1e8de8409d79f/stable/alpine/Dockerfile maybe its more clear to fork nginxinc/docker-nginx, make pull request and base our own vendoring on top of that?

Comment (ocraviotto) [inline Dockerfile]: @shirkevich Upgrading to 3.5 is not _longer_ the only change, removing unused modules (as suggested by @ilbambino ) and mixing instructions is a good idea too. I've tried to include only what we use downstream. Anyway, the goal with all of this was to finally base NGINX on our soon-to-come base image (still coupled in here, basically alpine plus dumb-init and secretary) which is mentioned in my NOTE2. BTW, after removing as many modules as I could make sense of removing the remaining CVE issues were gone.

**Pull request merged**